Path: utzoo!mnetor!uunet!lll-winken!lll-lcc!ames!elroy!devvax!lroot
From: lroot@devvax.JPL.NASA.GOV (The Superuser)
Newsgroups: comp.sources.bugs
Subject: perl 1.0 patch #17
Message-ID: <1226@devvax.JPL.NASA.GOV>
Date: 4 Feb 88 18:34:53 GMT
Organization: Jet Propulsion Laboratory, Pasadena, CA
Lines: 353
Summary: This is an official patch for perl 1.0.  Please apply it.

System: perl version 1.0
Patch #: 17
Priority: LOW
Subject: chmod 755 should be chmod +x; undefined signal handler now warns
From: gatech!turing!mike (Michael I. Bushnell)
From: tli@sargas.usc.edu (Tony Li)

Description:
	All chmod 755s should really be chmod +x to preserve group write
	permissions.

	Signal handlers without corresponding subroutine definitions now
	produce a warning message so that the programmer will know why
	his program is misbehaving.  It's still possible to define a null
	signal handler--you just have to have an explicit subroutine that
	does nothing.

Fix:	From rn, say "| patch -p0 -N -d DIR", where DIR is your perl source
	directory.  Outside of rn, say "cd DIR; patch -p0 -N <thisarticle".
	If you don't have the patch program, apply the following by hand,
	or get patch (version 2.0, latest patchlevel).

	If patch indicates that patchlevel is the wrong version, you may need
	to apply one or more previous patches, or the patch may already
	have been applied.  See the patchlevel.h file to find out what has or
	has not been applied.  In any event, don't continue with the patch.

	If you are missing previous patches they can be obtained from me:

	Larry Wall
	lwall@jpl-devvax.jpl.nasa.gov

	If you send a mail message of the following form it will greatly speed
	processing:

	Subject: Command
	@SH mailpatch PATH perl 1.0 LIST
		   ^ note the c

	where PATH is a return path FROM ME TO YOU in Internet notation, and
	LIST is the number of one or more patches you need, separated by spaces,
	commas, and/or hyphens.  Saying 35- says everything from 35 to the end.

	You can also get the patches via anonymous FTP from
	jpl-devvax.jpl.nasa.gov (128.149.8.43).

Index: patchlevel.h
Prereq: 16
1c1
< #define PATCHLEVEL 16
---
> #define PATCHLEVEL 17
 
Index: Makefile.SH
Prereq: 1.0.1.6
*** Makefile.SH.old	Thu Feb  4 10:27:26 1988
--- Makefile.SH	Thu Feb  4 10:27:27 1988
***************
*** 20,28 ****
  
  echo "Extracting Makefile (with variable substitutions)"
  cat >Makefile <<!GROK!THIS!
! # $Header: Makefile.SH,v 1.0.1.6 88/02/04 00:11:48 root Exp $
  #
  # $Log:	Makefile.SH,v $
  # Revision 1.0.1.6  88/02/04  00:11:48  root
  # patch16: forgot one ln -s.
  # 
--- 20,31 ----
  
  echo "Extracting Makefile (with variable substitutions)"
  cat >Makefile <<!GROK!THIS!
! # $Header: Makefile.SH,v 1.0.1.7 88/02/04 10:25:41 root Exp $
  #
  # $Log:	Makefile.SH,v $
+ # Revision 1.0.1.7  88/02/04  10:25:41  root
+ # patch17: changed chmod 755 to +x.
+ # 
  # Revision 1.0.1.6  88/02/04  00:11:48  root
  # patch16: forgot one ln -s.
  # 
***************
*** 126,136 ****
  	- if test `pwd` != $(bin); then cp $(public) $(bin); fi
  	cd $(bin); \
  for pub in $(public); do \
! chmod 755 `basename $$pub`; \
  done
  	- test $(bin) = /bin || rm -f /bin/perl
  	- test $(bin) = /bin || $(SLN) $(bin)/perl /bin || cp $(bin)/perl /bin
! #	chmod 755 makedir
  #	- makedir `filexp $(lib)`
  #	- \
  #if test `pwd` != `filexp $(lib)`; then \
--- 129,139 ----
  	- if test `pwd` != $(bin); then cp $(public) $(bin); fi
  	cd $(bin); \
  for pub in $(public); do \
! chmod +x `basename $$pub`; \
  done
  	- test $(bin) = /bin || rm -f /bin/perl
  	- test $(bin) = /bin || $(SLN) $(bin)/perl /bin || cp $(bin)/perl /bin
! #	chmod +x makedir
  #	- makedir `filexp $(lib)`
  #	- \
  #if test `pwd` != `filexp $(lib)`; then \
***************
*** 138,144 ****
  #fi
  #	cd `filexp $(lib)`; \
  #for priv in $(private); do \
! #chmod 755 `basename $$priv`; \
  #done
  	- if test `pwd` != $(mansrc); then \
  for page in $(manpages); do \
--- 141,147 ----
  #fi
  #	cd `filexp $(lib)`; \
  #for priv in $(private); do \
! #chmod +x `basename $$priv`; \
  #done
  	- if test `pwd` != $(mansrc); then \
  for page in $(manpages); do \
***************
*** 164,170 ****
  	./makedepend
  
  test: perl
! 	chmod 755 t/TEST t/base.* t/comp.* t/cmd.* t/io.* t/op.*
  	cd t && (rm -f perl; $(SLN) ../perl .) && ./perl TEST
  
  clist:
--- 167,173 ----
  	./makedepend
  
  test: perl
! 	chmod +x t/TEST t/base.* t/comp.* t/cmd.* t/io.* t/op.*
  	cd t && (rm -f perl; $(SLN) ../perl .) && ./perl TEST
  
  clist:
 
Index: x2p/Makefile.SH
Prereq: 1.0.1.1
*** x2p/Makefile.SH.old	Thu Feb  4 10:27:56 1988
--- x2p/Makefile.SH	Thu Feb  4 10:27:57 1988
***************
*** 18,26 ****
  esac
  echo "Extracting x2p/Makefile (with variable substitutions)"
  cat >Makefile <<!GROK!THIS!
! # $Header: Makefile.SH,v 1.0.1.1 88/01/26 14:15:24 root Exp $
  #
  # $Log:	Makefile.SH,v $
  # Revision 1.0.1.1  88/01/26  14:15:24  root
  # Added mallocsrc stuff.
  # 
--- 18,29 ----
  esac
  echo "Extracting x2p/Makefile (with variable substitutions)"
  cat >Makefile <<!GROK!THIS!
! # $Header: Makefile.SH,v 1.0.1.2 88/02/04 10:27:04 root Exp $
  #
  # $Log:	Makefile.SH,v $
+ # Revision 1.0.1.2  88/02/04  10:27:04  root
+ # patch17: changed chmod 755 to +x.
+ # 
  # Revision 1.0.1.1  88/01/26  14:15:24  root
  # Added mallocsrc stuff.
  # 
***************
*** 97,106 ****
  	- if test `pwd` != $(bin); then cp $(public) $(bin); fi
  	cd $(bin); \
  for pub in $(public); do \
! chmod 755 `basename $$pub`; \
  done
  	- test $(bin) = /bin || rm -f /bin/a2p
! #	chmod 755 makedir
  #	- makedir `filexp $(lib)`
  #	- \
  #if test `pwd` != `filexp $(lib)`; then \
--- 100,109 ----
  	- if test `pwd` != $(bin); then cp $(public) $(bin); fi
  	cd $(bin); \
  for pub in $(public); do \
! chmod +x `basename $$pub`; \
  done
  	- test $(bin) = /bin || rm -f /bin/a2p
! #	chmod +x makedir
  #	- makedir `filexp $(lib)`
  #	- \
  #if test `pwd` != `filexp $(lib)`; then \
***************
*** 108,114 ****
  #fi
  #	cd `filexp $(lib)`; \
  #for priv in $(private); do \
! #chmod 755 `basename $$priv`; \
  #done
  	- if test `pwd` != $(mansrc); then \
  for page in $(manpages); do \
--- 111,117 ----
  #fi
  #	cd `filexp $(lib)`; \
  #for priv in $(private); do \
! #chmod +x `basename $$priv`; \
  #done
  	- if test `pwd` != $(mansrc); then \
  for page in $(manpages); do \
 
Index: makedepend.SH
Prereq: 1.0.1.1
*** makedepend.SH.old	Thu Feb  4 10:27:33 1988
--- makedepend.SH	Thu Feb  4 10:27:34 1988
***************
*** 15,23 ****
  echo "Extracting makedepend (with variable substitutions)"
  $spitshell >makedepend <<!GROK!THIS!
  $startsh
! # $Header: makedepend.SH,v 1.0.1.1 88/02/02 11:24:05 root Exp $
  #
  # $Log:	makedepend.SH,v $
  # Revision 1.0.1.1  88/02/02  11:24:05  root
  # patch13: removed spurious -I./h.
  # 
--- 15,26 ----
  echo "Extracting makedepend (with variable substitutions)"
  $spitshell >makedepend <<!GROK!THIS!
  $startsh
! # $Header: makedepend.SH,v 1.0.1.2 88/02/04 10:26:06 root Exp $
  #
  # $Log:	makedepend.SH,v $
+ # Revision 1.0.1.2  88/02/04  10:26:06  root
+ # patch17: changed chmod 755 to +x.
+ # 
  # Revision 1.0.1.1  88/02/02  11:24:05  root
  # patch13: removed spurious -I./h.
  # 
***************
*** 145,151 ****
  
  !NO!SUBS!
  $eunicefix makedepend
! chmod 755 makedepend
  case `pwd` in
  *SH)
      $rm -f ../makedepend
--- 148,154 ----
  
  !NO!SUBS!
  $eunicefix makedepend
! chmod +x makedepend
  case `pwd` in
  *SH)
      $rm -f ../makedepend
 
Index: makedir.SH
Prereq: 1.0
*** makedir.SH.old	Thu Feb  4 10:27:41 1988
--- makedir.SH	Thu Feb  4 10:27:42 1988
***************
*** 15,23 ****
  echo "Extracting makedir (with variable substitutions)"
  $spitshell >makedir <<!GROK!THIS!
  $startsh
! # $Header: makedir.SH,v 1.0 87/12/18 13:05:32 root Exp $
  # 
  # $Log:	makedir.SH,v $
  # Revision 1.0  87/12/18  13:05:32  root
  # Initial revision
  # 
--- 15,26 ----
  echo "Extracting makedir (with variable substitutions)"
  $spitshell >makedir <<!GROK!THIS!
  $startsh
! # $Header: makedir.SH,v 1.0.1.1 88/02/04 10:26:18 root Exp $
  # 
  # $Log:	makedir.SH,v $
+ # Revision 1.0.1.1  88/02/04  10:26:18  root
+ # patch17: changed chmod 755 to +x.
+ # 
  # Revision 1.0  87/12/18  13:05:32  root
  # Initial revision
  # 
***************
*** 74,77 ****
  done
  !GROK!THIS!
  $eunicefix makedir
! chmod 755 makedir
--- 77,80 ----
  done
  !GROK!THIS!
  $eunicefix makedir
! chmod +x makedir
 
Index: stab.c
Prereq: 1.0.1.3
*** stab.c.old	Thu Feb  4 10:27:50 1988
--- stab.c	Thu Feb  4 10:27:51 1988
***************
*** 1,6 ****
! /* $Header: stab.c,v 1.0.1.3 88/02/04 00:13:15 root Exp $
   *
   * $Log:	stab.c,v $
   * Revision 1.0.1.3  88/02/04  00:13:15  root
   * patch16: stabset didn't copy some strings it should have.
   * 
--- 1,9 ----
! /* $Header: stab.c,v 1.0.1.4 88/02/04 10:26:31 root Exp $
   *
   * $Log:	stab.c,v $
+  * Revision 1.0.1.4  88/02/04  10:26:31  root
+  * patch17: now prints warning if no signal handler defined.
+  * 
   * Revision 1.0.1.3  88/02/04  00:13:15  root
   * patch16: stabset didn't copy some strings it should have.
   * 
***************
*** 285,298 ****
      STR *str;
  
      stab = stabent(str_get(hfetch(sigstab->stab_hash,sig_name[sig])),TRUE);
!     savearray = defstab->stab_array;
!     defstab->stab_array = anew();
!     str = str_new(0);
!     str_set(str,sig_name[sig]);
!     apush(defstab->stab_array,str);
!     str = cmd_exec(stab->stab_sub);
!     afree(defstab->stab_array);  /* put back old $_[] */
!     defstab->stab_array = savearray;
  }
  
  char *
--- 288,306 ----
      STR *str;
  
      stab = stabent(str_get(hfetch(sigstab->stab_hash,sig_name[sig])),TRUE);
!     if (stab->stab_sub) {
! 	savearray = defstab->stab_array;
! 	defstab->stab_array = anew();
! 	str = str_new(0);
! 	str_set(str,sig_name[sig]);
! 	apush(defstab->stab_array,str);
! 	str = cmd_exec(stab->stab_sub);
! 	afree(defstab->stab_array);  /* put back old $_[] */
! 	defstab->stab_array = savearray;
!     }
!     else
! 	fprintf(stderr,"perl warning: SIG%s handler \"%s\" not defined.\n",
! 	    sig_name[sig], stab->stab_name );
  }
  
  char *
